processor/internal/transformer/destination_transformer/transform.go

```diff
@@ -0,0 +1,38 @@
+package destination_transformer
+
+import (
+	"context"
+
+	"github.com/rudderlabs/rudder-server/processor/internal/transformer/destination_transformer/embedded/pubsub"
+	"github.com/rudderlabs/rudder-server/processor/types"
+)
+
+type transformerImpl func(ctx context.Context, clientEvents []types.TransformerEvent) types.Response
+
+var embeddedTransformerImpls = map[string]transformerImpl{
+	"GOOGLEPUBSUB": pubsub.Transform,
+}
+
+func (c *Client) Transform(ctx context.Context, clientEvents []types.TransformerEvent) types.Response {
```

Comment:
Struct methods should live on the same file as struct definition. In general opt for fewer files inside a package even if the file size is becoming larger.